src/Frontend/Core.hs

```diff
@@ -416,23 +416,24 @@ form formHandler = getH :<|> postH
 
 -- | Wrap anything that has 'ToHtml' and wrap it in an HTML body with complete page.
 data Frame body
-    = Frame { _frameUser :: User, _frameBody :: body }
+    = Frame { _frameUser :: User, _frameBody :: body, _frameMessages :: [StatusMessage] }
```

Comment:
After logout, we are on a public frame, and yet it makes sense to display a message in the frame like "thank you for playing".  But we can change that once we need it.